elements/pl-matching/pl-matching.py

```diff
@@ -157,17 +158,25 @@ def prepare(element_html, data):
             # Add a sample of the distractors.
             needed_options.extend(random.sample(distractors, more_needed))
```

Comment:
There is a big problem here for fixed order, since the distractors are always added at the end (same in line 153). This not only causes the order to deviate from documentation, but also gives a huge hint to students that know the behaviour, since they can focus on the top N options as correct when there are more options than statements, and assume the distractors are at the end.

Reply:
```suggestion
            distractor_sample = random.sample(distractors, more_needed)
            needed_options_keys = needed_options_keys.union(set(o['name'] for o in distractor_sample))
            needed_options = [o for o in options if o['name'] in needed_options_keys]
```

for line 153, `needed_options = options` probably resolves the issue.

Reply:
@nealterrell I have committed these changes, if you have a chance to test that would be great (I'd prefer a second set of eyes in these changes). Otherwise I tagged @mwest1066 and @nwalters512 to review it.

Reply:
I thought this might have been marked resolved by mistake so I unresolved it for visibility, but sorry if I was wrong to do that. Feel free to mark resolved again.